### ISSUE TITLE
smartypants incorrectly handling fractions correctly

### ISSUE THREAD

**Issue by SamSaffron** [vmg/redcarpet#506]
why all the hate for the 1 of April :smile: 

```
[11] pry(main)> Redcarpet::Render::SmartyPants.render("1/4 a year ago on 1/4/2010 something happened")
=> "&frac14; a year ago on &frac14;/2010 something happened"
```

should be 

```
=> "&frac14; a year ago on 1/4/2010 something happened"
```

there should be a bypass if the char after the fraction is `/` which is punctuation but not in that context

**Comment by SamSaffron** [comment]
fixed via #507

**Comment by robin850** [comment]
Thanks for reporting and fixing that fast! :+1:
